subsys/net/lib/mqtt_sn/mqtt_sn.c

```diff
@@ -126,13 +124,12 @@ static void mqtt_sn_con_init(struct mqtt_sn_confirmable *con)
 	con->last_attempt = 0;
 	con->retries = N_RETRY;
 	con->msg_id = next_msg_id();
-	con->in_use = true;
 }
 
 static void mqtt_sn_publish_destroy(struct mqtt_sn_client *client, struct mqtt_sn_publish *pub)
 {
 	sys_slist_find_and_remove(&client->publish, &pub->next);
-	memset(pub, 0, sizeof(*pub));
+	k_mem_slab_free(&pubs, (void *)&pub);
```

Comment:
Changing casting does not help here, you are still passing double pointer to the free func. You need to change this to `pub` and not its address.

Reply:
oh shit, I'm blind